src/OrchardCore.Modules/OrchardCore.Resources/ResourceManagementOptionsConfiguration.cs

```diff
@@ -1,29 +1,28 @@
 using Microsoft.AspNetCore.Http;
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Options;
-using OrchardCore.Environment.Shell;
 using OrchardCore.ResourceManagement;
+using OrchardCore.ResourceManagement.Core;
 using OrchardCore.Settings;
 
 namespace OrchardCore.Resources
 {
     public class ResourceManagementOptionsConfiguration : IConfigureOptions<ResourceManagementOptions>
     {
-        private readonly ISiteService _siteService;
+        private readonly IResourceSettingProvider _resourceSettingProvider;
```

Comment:
Don't introduce a new provider interface, it's unecesary, `IConfigureOptions` is already a provider based interface that can be registered multiple times.

Just introduce different implementations of `IConfigureOptions<ResourceManagementOptions>` that can provide the configuration from the different sources.

You could have two, one for building the manifests, and another for the default configuration from `ISiteService`

Reply:
Great approach! Done.